src/main/java/com/conveyal/datatools/manager/controllers/api/GtfsPlusController.java

```diff
@@ -73,9 +73,9 @@ public static Boolean uploadGtfsPlusFile (Request req, Response res) throws IOEx
         try {
             uploadStream = part.getInputStream();
             gtfsPlusStore.newFeed(feedVersionId, uploadStream, null);
-        } catch (Exception e) {
+        } catch (IOException e) {
             LOG.error("Unable to open input stream from upload");
-            haltWithMessage(req, 500, "an unexpected error occurred", e);
+            haltWithMessage(req, 400, "Unable to open input stream from upload", e);
```

Comment:
Related to the early comment about having `haltWithMessage` also log messages. This would keep things DRY-er here.